src/main/java/core/levels/LevelManager.java

```diff
@@ -26,12 +30,86 @@ public LevelManager(LevelState level) {
         map = new TmxMapLoader().load(level.getMapPath());
         this.mapRenderer = new OrthogonalTiledMapRenderer(map, level.getUnitScale());
         this.worldManager = new WorldManager(level.world);
-
         this.batch = new SpriteBatch();
+
+        // assign all enemies to current worldManager
+        List<EntitySpawner> enemiesUpdated = level.getEnemySpawns();
+        for (EntitySpawner enemy : enemiesUpdated){ enemy.setWorldManager(this.worldManager);}
+        level.setEnemySpawns(enemiesUpdated);
     }
 
+    /**
+     * Step physics simulation of World
+     * Elapse time in World
+     * Spawn enemies in World
+     *
+     * Above actions only performed if level isn't in a paused state
+     *
+     * @param dt time delta to simulate (seconds) (capped at .5 in case computer is too slow)
+     */
     public void step(float dt) {
+        // If level isn't paused, perform step, time elapsing, spawning
+        if (level.getLevelPaused()) {
+        return; }
+
+        // Stepping physics simulation
         worldManager.step(dt);
+
+        // Elapsing time in world
+        level.setCurrentTime(level.getCurrentTime() + dt);
+
+        // Spawning enemies in world
+        updateEnemies();
+    }
+
+    /**
+     * Periodically spawn enemies into the world, based on spawnTime given in LevelState
+     * If all enemies spawned, check if game has been won
+     */
+    private void updateEnemies() {
+        // Spawning enemies in world
+        List<EntitySpawner> enemies = level.getEnemySpawns();
+
+        if (enemies.isEmpty()){
+            // If all enemies have been spawned, check if game is won or not
+            if (checkWin()){showWinCondition();} // Roy will implement showWinCondition()
+        }
+        else {
+            // Spawn enemy every 15 seconds
+            if (level.getCurrentTime() >= level.getSpawnTime()) {
+                EntitySpawner enemy = enemies.remove(0);
+                enemy.spawn();
+                level.setEnemySpawns(enemies);
+                level.setScore(level.getScore() + 1);
+                level.setSpawnTime(level.getSpawnTime() + 15);
+            }
+        }
+    }
+
+    /**
+     * Checking win condition
+     * level is won if 100 seconds have passed
+     *
+     * @return whether win condition has been met or not
+     */
+    public boolean checkWin(){
+        return level.getCurrentTime() >= 100;
+    }
+
+    /**
+     * Save current time elapsed from level
+     * @throws IOException relating to savedState.txt
+     * @throws ClassNotFoundException relating to reading objects in
+     */
+    public void saveState() throws IOException, ClassNotFoundException {
```

Comment:
Where will this save? If possible, I think it would make sense to save within a separate folder for cleanness and to allow us to read easier if we want to have multiple save states - may also be something for phase 2